packages/dev/core/src/node.ts

```diff
@@ -814,6 +814,38 @@ export class Node implements IBehaviorAware<Node> {
         return this._ranges[name] || null;
     }
 
+    /**
+     * Clone the current node
+     * @param name Name of the new clone
+     * @param newParent New parent for the clone
+     * @param doNotCloneChildren Do not clone children hierarchy
+     * @returns the new transform node
+     */
+    public clone(name: string, newParent: Nullable<Node>, doNotCloneChildren?: boolean): Nullable<Node> {
+        const result = SerializationHelper.Clone(() => new Node(name, this.getScene()), this);
+
+        result.name = name;
+        result.id = name;

```

Comment:
These two assignments are already done in the `Node` constructor.

Reply:
Clone in TransformNode returns Nullable, so I just copied the same method. But I think there's no problem in taking Nullable out.